sonar-python-plugin/src/main/java/org/sonar/plugins/python/SensorTelemetryStorage.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+package org.sonar.plugins.python;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.Version;
+
+public class SensorTelemetryStorage {
+  private static final Logger LOG = LoggerFactory.getLogger(SensorTelemetryStorage.class);
+
+  private final Map<String, String> data;
+
+  public SensorTelemetryStorage() {
+    data = new HashMap<>();
+  }
+
+  public Map<String, String> data() {
+    return Collections.unmodifiableMap(data);
+  }
+
+  public void send(SensorContext sensorContext) {
+    // This try/catch block should be useless, as in the worst case it should be a no-op depending on the SensorContext implementation
+    // It exists to be extra sure for the LTA
+    try {
+      var apiVersion = sensorContext.runtime().getApiVersion();
+      if (apiVersion.isGreaterThanOrEqual(Version.create(10, 9))) {
+        data.forEach((k, v) -> {
+          LOG.info("Collected metric: {}={}", k, v);
+          sensorContext.addTelemetryProperty(k, v);
+        });
+
+      } else {
+        LOG.info("Skipping sending metrics because the plugin API version is {}", apiVersion);
+      }
+    } catch (Exception e) {
+      LOG.error("Failed to send metrics", e);
+    }
+  }
+
+  public void updateMetric(MetricKey key, String value) {
+    data.put(key.key(), value);
+  }
+
+  public enum MetricKey {
```

Comment:
Since this enum massively used outside of the `SensorTelemetryStorage` - probably it makes sense to make it a top-level class